src/_common/community/activity-item/AppCommunityActivityItem.vue

```diff
@@ -17,194 +17,179 @@ import { CommunityCompetitionModel } from '../competition/competition.model';
-
-	get loggedOn() {
-		return formatDate(this.item.added_on, 'medium');
-	}
-
-	get shouldShowIcon() {
-		return !!this.icon && this.showIcon;
-	}
-
-	get icon() {
-		return this.item.getTypeIcon()?.icon;
-	}
-
-	get color() {
-		return this.item.getTypeIcon()?.color;
-	}
-
-	get actionIsFiresidePost() {
-		return this.item.action_resource instanceof FiresidePostModel;
-	}
-
-	get actionIsUser() {
-		return this.item.action_resource instanceof UserModel;
-	}
-
-	get isToday() {
-		return (
-			formatDate(this.item.added_on, 'mediumDate') === formatDate(Date.now(), 'mediumDate')
-		);
+	showIcon: {
+		type: Boolean,
+		required: true,
+	},
+});
+
+const CommunityCreated = CommunityActivityItemType.CommunityCreated;
+const PostFeature = CommunityActivityItemType.PostFeature;
+const PostUnfeature = CommunityActivityItemType.PostUnfeature;
+const PostMove = CommunityActivityItemType.PostMove;
+const PostEject = CommunityActivityItemType.PostEject;
+
+const ModInvite = CommunityActivityItemType.ModInvite;
+const ModAccept = CommunityActivityItemType.ModAccept;
+const ModRemove = CommunityActivityItemType.ModRemove;
+
+const BlockUser = CommunityActivityItemType.BlockUser;
+
+const EditDescription = CommunityActivityItemType.EditDescription;
+const EditThumbnail = CommunityActivityItemType.EditThumbnail;
+const EditHeader = CommunityActivityItemType.EditHeader;
+const EditDetails = CommunityActivityItemType.EditDetails;
+const EditHeaderRemove = CommunityActivityItemType.EditHeaderRemove;
+
+const ChannelAdd = CommunityActivityItemType.ChannelAdd;
+const ChannelRemove = CommunityActivityItemType.ChannelRemove;
+const ChannelEdit = CommunityActivityItemType.ChannelEdit;
+const ChannelRename = CommunityActivityItemType.ChannelRename;
+
+const GameLink = CommunityActivityItemType.GameLink;
+const GameUnlink = CommunityActivityItemType.GameUnlink;
+
+const CompetitionEditSettings = CommunityActivityItemType.CompetitionEditSettings;
+const CompetitionEditVoting = CommunityActivityItemType.CompetitionEditVoting;
+const CompetitionVotingSetActive = CommunityActivityItemType.CompetitionVotingSetActive;
+const CompetitionEntryRemove = CommunityActivityItemType.CompetitionEntryRemove;
+const CompetitionEntryUnremove = CommunityActivityItemType.CompetitionEntryUnremove;
+const CompetitionEntryGiveAward = CommunityActivityItemType.CompetitionEntryGiveAward;
+
+const FiresideStart = CommunityActivityItemType.FiresideStart;
+const FiresideStartDraft = CommunityActivityItemType.FiresideStartDraft;
+const FiresidePublish = CommunityActivityItemType.FiresidePublish;
+const FiresideExtinguish = CommunityActivityItemType.FiresideExtinguish;
+const FiresideFeature = CommunityActivityItemType.FiresideFeature;
+const FiresideUnfeature = CommunityActivityItemType.FiresideUnfeature;
+const FiresideEject = CommunityActivityItemType.FiresideEject;
+
+const { item, showIcon } = toRefs(props);
+
+const icon = computed(() => {
+	return item.value.getTypeIcon()?.icon;
```

Comment:
we can change the typing of this icon to fix the typescript warning in the template, but it's hard to say how in a comment, let's discuss on slack